pkg/xds/topology/outbound.go

```diff
@@ -154,6 +154,48 @@ func fillDataplaneOutbounds(
 	}
 }
 
+func BuildCrossMeshEndpointMap(
+	mesh *core_mesh.MeshResource,
+	zone string,
+	gateways []*core_mesh.MeshGatewayResource,
+	dataplanes []*core_mesh.DataplaneResource,
+	zoneIngresses []*core_mesh.ZoneIngressResource,
+) core_xds.EndpointMap {
+	outbound := core_xds.EndpointMap{}
+	for _, dataplane := range dataplanes {
+		gateway := SelectGateway(gateways, dataplane.Spec.Matches)
+		if !dataplane.Spec.IsBuiltinGateway() {
+			continue
+		}
+		dpSpec := dataplane.Spec
+		dpNetworking := dpSpec.GetNetworking()
+
+		dpGateway := dpNetworking.GetGateway()
+		dpTags := dpGateway.GetTags()
+		serviceName := dpTags[mesh_proto.ServiceTag]
+
+		for _, listener := range gateway.Spec.Conf.Listeners {
+			outbound[serviceName] = append(outbound[serviceName], core_xds.Endpoint{
+				Target: dpNetworking.GetAddress(),
+				Port:   listener.GetPort(),
+				Tags:   mesh_proto.Merge(dpTags, gateway.Spec.GetTags(), listener.GetTags()),
+				Weight: 1,
+				// TODO Locality: ,
+			})
+		}
+	}
+	// TODO handle the endpoint weight thing
+	core.Log.Info("outbounds", "mesh", mesh, "zone", zone, "ingresses", zoneIngresses)
+	_ = fillIngressOutbounds(
+		outbound,
+		zoneIngresses,
+		nil,
```

Comment:
should egresses be nil here?

Reply:
No, but I haven't tackled ZoneEgress yet. It's likely I will even remove the commit adding ZoneIngress support for this first PR.

Reply:
Will look at ingress/multizone stuff in next pr